server/controllers/bookingRequest.js

```diff
@@ -0,0 +1,81 @@
+const asyncHandler = require("express-async-handler");
+const User = require("../models/User");
+const BookingRequest = require("../models/BookingRequest");
+
+// @route POST /booking-requests/
+// @desc Create a new request
+// @access Private
+
+exports.createBookingRequest = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+  const user = await User.findById(userId);
+  const { sitterId, start, end, description } = req.body;
```

Comment:
Optional: Along with destructuring the required fields you can also validate the request body so that none of the required fields are missing.

```
if (!sitterId | !start.....) {
    return res.status(400).send("Bad Request");
}
```

Reply:
I see - I had assumed that the client might only send the data that it wanted to update, but should I assume that the client will always send all of the fields regardless of whether it has be changed?